lib/wallet/src/Cardano/Wallet/DB/Store/CBOR/Store.hs

```diff
@@ -41,39 +59,45 @@ import Database.Persist
 import Database.Persist.Sql
     ( SqlPersistT )
 
-import qualified Cardano.Wallet.DB.Sqlite.Schema as Schema
-    ( CBOR (..), EntityField (..) )
+import qualified Data.ByteString.Lazy as BL
 import qualified Data.Map.Strict as Map
 
-txCBORiso :: Iso' (TxId, TxCBOR) Schema.CBOR
-txCBORiso = iso f g
-  where
-    f :: (TxId, TxCBOR) -> Schema.CBOR
-    f (id',TxCBOR{..}) =
-        Schema.CBOR id' (toStrict txCBOR) (fromIntegral $ fromEnum txEra)
-    g :: Schema.CBOR -> (TxId, TxCBOR)
-    g Schema.CBOR{..} =
-        ( cborTxId
-        , TxCBOR (fromStrict cborTxCBOR) (toEnum $ fromIntegral cborTxEra)
-        )
+type TxCBORRaw = (BL.ByteString, Int)
+
+i :: Iso' (BL.ByteString, Int) (ByteString, Word16)
+i = iso (toStrict *** fromIntegral) (fromStrict *** fromIntegral)
+
+toTxCBOR :: (TxId, TxCBOR) -> (CBOR, TxCBORRaw)
+toTxCBOR (id', tx) = let
+        r = build eraValueSerialize tx
+        in (uncurry (CBOR id') $ r ^. i, r)
```

Comment:
```suggestion
toTxCBOR (id', tx) =
        let r = build eraValueSerialize tx
        in  (uncurry (CBOR id') $ r ^. i, r)
```

(Formatting: `let` in front of the line is a slightly easier to parse visually.)